### ISSUE TITLE
Packer produces damaged tar archives

### ISSUE THREAD

**Issue by kelseyhightower** [mitchellh/packer#2131]
Packer version: Packer v0.7.5
OS: OS X 10.10.3

Build a VMware image:

```
packer build -var-file settings.json kubestack-solo.json
```

### Template file

```
{
  "variables": {
    "kubestack_file": "{{env `KUBESTACK_FILE`}}",
    "source_path": "{{env `SOURCE_PATH`}}",
    "ssh_username": "{{env `SSH_USERNAME`}}",
    "ssh_key_path": "{{env `SSH_KEY_PATH`}}"
  },
  "builders": [{
    "type": "vmware-vmx",
    "headless": true,
    "source_path": "{{user `source_path`}}",
    "ssh_username": "{{user `ssh_username`}}",
    "ssh_key_path": "{{user `ssh_key_path`}}",
    "shutdown_command": "sudo poweroff",
    "vm_name": "kubestack-solo"
  }],
  "provisioners": [{
        "type": "file",
        "source": "static",
        "destination": "/tmp"
    },
    {
        "type": "shell",
        "inline": [
            "sudo mkdir -p /etc/kubernetes",
            "sudo mkdir -p /opt/bin",
            "sudo tar -xvf /tmp/static/{{user `kubestack_file`}} -C /opt/bin",
            "sudo chmod +x -R /opt/bin",
            "sudo chown root:root -R /opt/bin",
            "sudo mv /tmp/static/*.service /etc/systemd/system",
            "sudo mv /tmp/static/kubernetes.env /etc/kubernetes.env",
            "sudo mv /tmp/static/node.json /etc/kubernetes/node.json",
            "sudo systemctl enable etcd",
            "sudo systemctl enable kube-apiserver",
            "sudo systemctl enable kube-controller-manager",
            "sudo systemctl enable kube-scheduler",
            "sudo systemctl enable docker",
            "sudo systemctl enable kube-kubelet",
            "sudo systemctl enable kube-proxy",
            "rm -rf /tmp/static"
        ]
    }],
    "post-processors": [{
        "type": "compress",
        "output": "kubestack-solo-vmware.tar.gz"
    }]
}
```

### Errors

Try to view or open the resulting compressed tar:

```
tar -tzvf kubestack-solo-vmware.tar.gz 
-rw-------  0 501    20  742850560 May 22 10:26 Virtual Disk-cl1.vmdk
-rw-------  0 501    20       8684 May 22 10:26 kubestack-solo.nvram
tar: Damaged tar archive
tar: Retrying...
tar: Damaged tar archive
tar: Retrying...
tar: Damaged tar archive
tar: Retrying...
tar: Damaged tar archive
tar: Retrying...
tar: Damaged tar archive
tar: Retrying...
tar: Damaged tar archive
tar: Retrying...
tar: Damaged tar archive
tar: Retrying...
tar: Damaged tar archive
tar: Retrying...
tar: Damaged tar archive
tar: Retrying...
tar: Damaged tar archive
tar: Retrying...
```

**Comment by okeefm** [comment]
I was able to duplicate this in my Packerfile. To fix it, you can add `"format": "tar.gz"` to your post-processor. It's shown in the page on post-processors (https://www.packer.io/docs/templates/post-processors.html) but not on the compressor post-processor page specifically.

**Comment by mitchellh** [comment]
Dup of #2060 

We should be able to fix this with #2119
